pkg/publicshare/manager/owncloudsql/owncloudsql_test.go

```diff
@@ -0,0 +1,488 @@
+// Copyright 2018-2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package owncloudsql_test
+
+import (
+	"context"
+	"database/sql"
+	"os"
+	"strings"
+	"time"
+
+	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	link "github.com/cs3org/go-cs3apis/cs3/sharing/link/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	typespb "github.com/cs3org/go-cs3apis/cs3/types/v1beta1"
+	conversions "github.com/cs3org/reva/v2/internal/http/services/owncloud/ocs/conversions"
+	ctxpkg "github.com/cs3org/reva/v2/pkg/ctx"
+	"github.com/cs3org/reva/v2/pkg/publicshare"
+	"github.com/cs3org/reva/v2/pkg/publicshare/manager/owncloudsql"
+	"github.com/cs3org/reva/v2/pkg/share/manager/owncloudsql/mocks"
+	"github.com/cs3org/reva/v2/pkg/utils"
+	"github.com/stretchr/testify/mock"
+
+	_ "github.com/mattn/go-sqlite3"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("SQL manager", func() {
+	var (
+		m    publicshare.Manager
+		user *userpb.User
+		ctx  context.Context
+
+		ri    *provider.ResourceInfo
+		grant *link.Grant
+		// share *link.PublicShare
+
+		testDbFile *os.File
+		sqldb      *sql.DB
+	)
+
+	AfterEach(func() {
+		os.Remove(testDbFile.Name())
+	})
+
+	BeforeEach(func() {
+		var err error
+		testDbFile, err = os.CreateTemp("", "example")
+		Expect(err).ToNot(HaveOccurred())
+
+		dbData, err := os.ReadFile("test.db")
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = testDbFile.Write(dbData)
+		Expect(err).ToNot(HaveOccurred())
+		err = testDbFile.Close()
+		Expect(err).ToNot(HaveOccurred())
+
+		sqldb, err = sql.Open("sqlite3", testDbFile.Name())
+		Expect(err).ToNot(HaveOccurred())
+
+		userConverter := &mocks.UserConverter{}
+		userConverter.On("UserIDToUserName", mock.Anything, mock.Anything).Return("username", nil)
+		userConverter.On("UserNameToUserID", mock.Anything, mock.Anything).Return(
+			func(_ context.Context, username string) *userpb.UserId {
+				return &userpb.UserId{
+					OpaqueId: username,
+				}
+			},
+			func(_ context.Context, username string) error { return nil })
```

Comment:
```suggestion
			nil
```
should do if I'm not mistaken.